<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Auxiliary - Bioinformatics Project Meeting Agenda • prolfqua</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Auxiliary - Bioinformatics Project Meeting Agenda">
<meta property="og:description" content="prolfqua">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">prolfqua</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Auxiliary_ExDesignSurvey.html">Auxiliary - Bioinformatics Project Meeting Agenda</a>
    </li>
    <li>
      <a href="../articles/Auxiliary_prolfquaAbstract_2.html">Auxiliary - Euro Bioconductor 2020</a>
    </li>
    <li>
      <a href="../articles/Auxiliary_prolfquaAbstract_BioC2021.html">Auxiliary - Prolfqua - R package for visualization and modelling of proteomics label-free quantification data</a>
    </li>
    <li>
      <a href="../articles/BenchmarkingIonstarData.html">Benchmarking normalization, aggregation and models using the Ionstar Dataset</a>
    </li>
    <li>
      <a href="../articles/BenchmarkMSFraggerProteinIonStar.html">Benchmarking MSFragger output using Ionstar Dataset</a>
    </li>
    <li>
      <a href="../articles/Benchmark_MSStats.html">Benchmarking MSstats using the Ionstar Dataset</a>
    </li>
    <li>
      <a href="../articles/Benchmark_proDA_fromMQlfq.html">Benchmarking the proDA package using the Ionstar Dataset MQ LFQ intensities</a>
    </li>
    <li>
      <a href="../articles/Benchmark_proDA_medpolish.html">Benchmarking the proDA package using the Ionstar Dataset starting from peptides</a>
    </li>
    <li>
      <a href="../articles/Comparing2Groups.html">Comparing Two Groups</a>
    </li>
    <li>
      <a href="../articles/CreatingConfigurations.html">Creating prolfqua Configurations</a>
    </li>
    <li>
      <a href="../articles/Modelling2Factors.html">Modelling Dataset with two Factors.</a>
    </li>
    <li>
      <a href="../articles/Preprocessing2Factors.html">Preprocessing Dataset with two Factors.</a>
    </li>
    <li>
      <a href="../articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a>
    </li>
    <li>
      <a href="../articles/QualityControlAndSampleSizeEstimation.html">QC and Sample Size Estimation</a>
    </li>
    <li>
      <a href="../articles/Supplement_srm_summary.html">Supplement - Summarize SRM Peptide Level Measurements</a>
    </li>
    <li>
      <a href="../articles/Supplement_srm_summary_HL.html">Supplement - Summarize heavy light Peptide Level Measurements</a>
    </li>
    <li>
      <a href="../articles/Supplement_TidyAnalysis_HL_Skyline.html">Supplement - Data preprocessing of SRM Heavy Light data</a>
    </li>
    <li>
      <a href="../articles/Supplement_TidyAnalysis_Skyline_PRM.html">Supplement - Data preprocessing for PRM</a>
    </li>
    <li>
      <a href="../articles/Supplement_TidyCorrelationAnalysis.html">Supplement - Filter precursors or fragments based on correlation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wolski/prolfqua/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Auxiliary - Bioinformatics Project Meeting Agenda</h1>
                        <h4 class="author">FGCZ - (Draft)</h4>
            
            <h4 class="date">04 November, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wolski/prolfqua/blob/master/../vignettes/Auxiliary_ExDesignSurvey.Rmd"><code>../vignettes/Auxiliary_ExDesignSurvey.Rmd</code></a></small>
      <div class="hidden name"><code>Auxiliary_ExDesignSurvey.Rmd</code></div>

    </div>

    
    

<p>Project Nr :</p>
<p>Order Id :</p>
<p>Participants:</p>
<ul>
<li>User :</li>
<li>Bioinformatician :</li>
<li>Coach :</li>
</ul>
<p>When?</p>
<ul>
<li>
<input type="checkbox" disabled>
before start of experiment</li>
<li>
<input type="checkbox" disabled>
after QC sample preparation</li>
<li>
<input type="checkbox" disabled>
after main experiment sample preparation</li>
</ul>
<p>Comments :</p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<div id="overview-of-the-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#overview-of-the-analysis" class="anchor"></a>Overview of the analysis</h1>
<p>In label-free quantification (LFQ) experiments, we measure relative protein changes among groups of subjects. Before the main experiment, a quality control (QC) experiment is performed. We use the data from the QC to estimate the <em>within-group</em> variance and to determine the number of samples in the main experiment. The measurements of the main experiment are modeled using linear models, mixed-effects linear models, which can be thought of as an extension of the t-test to more than two groups. These estimates are used to perform Gene Set Enrichment Analysis and Overrepresentation analysis using WebGestaltR and sigora. The meeting aims to obtain and document all the information relevant to perform a statistically sound LFQ data analysis.</p>
</div>
<div id="aim-of-the-meeting" class="section level1">
<h1 class="hasAnchor">
<a href="#aim-of-the-meeting" class="anchor"></a>Aim of the meeting</h1>
<p>The goal of the meeting is this protocol specifying:</p>
<ul>
<li>which protein database to use</li>
<li>which and how many samples to use for the QC</li>
<li>determine all the parameters for the sample size estimation</li>
<li>the design of your main experiment</li>
<li>all the names of the factors and factor levels</li>
<li>specify the hypothesis to be tested</li>
</ul>
<p>The samples should be annotated with the names specified in the protocol in the b-fabric system. These names will also appear in the result tables and visualizations.</p>
<p>We perform the data analysis according to this protocol. Changing the analysis protocol (e.g. adding contrasts or blocking factors) after the analysis was delivered will incur time delays and additional costs.</p>
</div>
<div id="genomics-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#genomics-analysis" class="anchor"></a>Genomics Analysis</h1>
<p>Did you perform a genomics experiment on the same samples?</p>
<ul>
<li>
<input type="checkbox" disabled>
no</li>
<li>
<input type="checkbox" disabled>
yes</li>
</ul>
<p>What type of experiment?</p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<ul>
<li>Was it run at the FGCZ?</li>
<li>project Id:</li>
<li>order Id:</li>
<li>Bioinformatician :</li>
<li>Are you interested in Proteomics Genomics data integration?</li>
</ul>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p>Note:</p>
<ul>
<li>NGS data can be used to create improved protein identification databases.</li>
<li>If NGS - proteomics data integration is planned, compatible statistical analysis should be applied to the NGS and Proteomics data.</li>
</ul>
</div>
<div id="protein-sequence-database-and-downstream-analysis-tools" class="section level1">
<h1 class="hasAnchor">
<a href="#protein-sequence-database-and-downstream-analysis-tools" class="anchor"></a>Protein sequence database and downstream analysis tools</h1>
<p>Which sequence database should be used for protein identification?</p>
<p>Which organism?</p>
<ul>
<li>
<input type="checkbox" disabled>
homo sapiens</li>
<li>
<input type="checkbox" disabled>
mus musculus</li>
<li>
<input type="checkbox" disabled>
yeast</li>
<li>
<input type="checkbox" disabled>
zebra-fish</li>
<li>other :</li>
</ul>
<p>Should additional protein sequences be added to the database?</p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p>Contaminant list?</p>
<ul>
<li>
<input type="checkbox" disabled>
Complete</li>
<li>
<input type="checkbox" disabled>
Without human contaminants</li>
<li>other :</li>
</ul>
<p>Which downstream analysis tools will be used?</p>
<ul>
<li>
<input type="checkbox" disabled><a href="http://www.webgestalt.org/" target="_blank">WebGestalt</a> (WEB-based GEne SeT AnaLysis Toolkit)</li>
<li>
<input type="checkbox" disabled><a href="https://string-db.org/" target="_blank">string-db</a> (Known and predicted protein-protein interactions)</li>
<li>other :</li>
</ul>
<p><span class="math inline">\(~\)</span></p>
<p>Note:</p>
<ul>
<li><p>Typically we are using protein databases from uniprot.org</p></li>
<li><p>Ideally the identifiers in the protein database are compatible with the downstream analysis tool you intend to use. For instance <em>WebGestalt</em>, although it recognizes many types of identifiers: e.g., Uniprot Swiss-Prot, Gene Symbols, NCBI Entrez gene, it does not work with Uniprot Treml identifiers.</p></li>
</ul>
</div>
<div id="effect-size-and-size-of-test" class="section level1">
<h1 class="hasAnchor">
<a href="#effect-size-and-size-of-test" class="anchor"></a>Effect size and size of test</h1>
<p>These parameters are crucial for the QC experiment and will be assessed again when protein variance estimates from the QC experiment are available. They also will be used when visualizing the results in the final report (e.g., Volcano plots) or when running overrepresentation analysis (ORA).</p>
<div id="effect-size" class="section level2">
<h2 class="hasAnchor">
<a href="#effect-size" class="anchor"></a>Effect size</h2>
<p>What is the smallest effect size (i.e., fold change in protein abundance) of biological relevance, you are interested in detecting? Typically fold changes are <span class="math inline">\(1.5\)</span>, <span class="math inline">\(2\)</span>, or <span class="math inline">\(4\)</span>. This information is needed to determine the sample size as well as to biologically interpret the data.</p>
<p>The number of samples you need to use highly depends on the effect size you want to detect. Detecting small effect sizes requires you to measure more samples. On the other hand, the statistical significance of a fold change does not imply biological relevance. Consequently, do not measure more samples than is required.</p>
<ul>
<li>
<input type="checkbox" disabled>
1.5</li>
<li>
<input type="checkbox" disabled>
2</li>
<li>
<input type="checkbox" disabled>
4</li>
<li>other :</li>
</ul>
</div>
<div id="size-of-the-test5" class="section level2">
<h2 class="hasAnchor">
<a href="#size-of-the-test5" class="anchor"></a>Size of the test<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h2>
<p>Typically a difference between two conditions is considered to be significant if the p-Value is less than <span class="math inline">\(0.05\)</span>. For some applications a less significant p-Value can be used (e.g., <span class="math inline">\(0.1\)</span>) for others a smaller p-Value might be required.</p>
<ul>
<li>
<input type="checkbox" disabled>
0.01</li>
<li>
<input type="checkbox" disabled>
0.05</li>
<li>
<input type="checkbox" disabled>
0.1</li>
<li>other :</li>
</ul>
</div>
<div id="power-of-test" class="section level2">
<h2 class="hasAnchor">
<a href="#power-of-test" class="anchor"></a>Power of test</h2>
<p>The power of a binary hypothesis test is the probability that the test rejects the null hypothesis (H0) when a specific alternative hypothesis (H1) is true <a href="https://en.wikipedia.org/wiki/Power_(statistics)">Wikipedia</a></p>
<ul>
<li>
<input type="checkbox" disabled>
0.8</li>
<li>other :</li>
</ul>
</div>
<div id="does-the-variance-within-groups-differ" class="section level2">
<h2 class="hasAnchor">
<a href="#does-the-variance-within-groups-differ" class="anchor"></a>Does the variance within groups differ?</h2>
<p>Is the variance within the groups the same or different? E.g., do you expect some groups with larger within group differences?</p>
<ul>
<li>
<input type="checkbox" disabled>
yes</li>
<li>
<input type="checkbox" disabled>
no</li>
</ul>
<p>If yes, which of the group’s might have the highest within-group variance? (Some treatment might have heterogeneous outcomes.)</p>
<p><span class="math inline">\(~\)</span> <span class="math inline">\(~\)</span></p>
<p>If possible, use samples from this group for the QC experiment, which will enable you to get a conservative estimate of the required samples sizes. If the variance in one group is substantially higher, the power of the tests can be improved by allocating more samples to this group.</p>
</div>
<div id="difference-in-protein-variances" class="section level2">
<h2 class="hasAnchor">
<a href="#difference-in-protein-variances" class="anchor"></a>Difference in protein variances</h2>
<p>From the QC experiment, we can estimate for each protein a within-group variance. For some proteins, we observe more variation within a group compared to other proteins.</p>
<p>Are there proteins or protein groups of particular interest?</p>
<p>Provide protein IDs:</p>
<p>Do you want to obtain significant p-Values for proteins with large within-group variances, or would you be satisfied with getting significances for the <span class="math inline">\(X\%\)</span> of low variance proteins?</p>
<ul>
<li>
<input type="checkbox" disabled><span class="math inline">\(50\%\)</span>
</li>
<li>
<input type="checkbox" disabled><span class="math inline">\(70\%\)</span>
</li>
<li>other :</li>
</ul>
<p>Note:</p>
<ul>
<li>High abundant proteins will exhibit smaller technical variability compared with low abundant proteins.</li>
</ul>
</div>
</div>
<div id="designs" class="section level1">
<h1 class="hasAnchor">
<a href="#designs" class="anchor"></a>Designs</h1>
<p>This section covers the most frequently used designs.</p>
<ul>
<li><p><input type="checkbox" disabled>
Parallel Group Design - Evaluates a single factor<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p></li>
<li><p><input type="checkbox" disabled>
Factorial Design - Evaluates multiple factors simultaneously</p></li>
</ul>
<div id="parallel-group-design" class="section level2">
<h2 class="hasAnchor">
<a href="#parallel-group-design" class="anchor"></a>Parallel Group Design</h2>
<p>How many groups?</p>
<ul>
<li>
<input type="checkbox" disabled>
2</li>
<li>
<input type="checkbox" disabled>
more. How many?</li>
</ul>
<p>Factor name (<em>camel case</em><a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> e.g., Treatment) :</p>
<p>Level names (<em>camel case</em> e.g. Control, Diet, Creme):</p>
</div>
<div id="factorial-design8" class="section level2">
<h2 class="hasAnchor">
<a href="#factorial-design8" class="anchor"></a>Factorial Design<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>
</h2>
<p>Applies when studying how two or more factors influence protein expression, e.g., <span class="math inline">\(Knockout\)</span> and <span class="math inline">\(Treatment\)</span>.</p>
<ul>
<li>How many factors? Typically 2 rarely 3.</li>
<li>How many levels per factor? Typically 2 or 3 per factor.</li>
</ul>
<table class="table">
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
<th align="center">level names <span class="math inline">\(~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\)</span>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
<td align="center"></td>
</tr>
</tbody>
</table>
<ul>
<li>Provide computer-friendly names of the factors (grouping variables) in <em>camel case</em>.</li>
<li>Provide names of levels for each grouping variable.</li>
</ul>
<p>Example:</p>
<table class="table">
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
<th align="center">level names <span class="math inline">\(~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\)</span>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Treatment</td>
<td>3</td>
<td align="center">Control, Diet, Creme</td>
</tr>
<tr class="even">
<td align="left">AgeGroup</td>
<td>2</td>
<td align="center">young, old</td>
</tr>
</tbody>
</table>
</div>
<div id="repeated-measurements" class="section level2">
<h2 class="hasAnchor">
<a href="#repeated-measurements" class="anchor"></a>Repeated measurements</h2>
<p>Applies when measurements are repeated on the same subject, e.g. over time <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<ul>
<li>
<input type="checkbox" disabled>
Yes</li>
<li>
<input type="checkbox" disabled>
No</li>
</ul>
<p>If <em>yes</em> please specify a computer-friendly factor name using <em>camel case</em>: e.g., PatientId.</p>
<p>Name of the factor :</p>
<p>The subject identifiers can contain upper and lower case letters and numbers but avoid special characters and white spaces. Neither the factor names or the levels of the blocking factors will be shown in the results or figures generated.</p>
</div>
<div id="blocking-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#blocking-factors" class="anchor"></a>Blocking factors</h2>
<p>A blocking factor is some variable that affects an experimental outcome but is itself of no interest, e.g., sample batch <a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a>. Including blocking-factors can improve the statistical model.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
</tr>
<tr class="odd">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><span class="math inline">\(~\)</span></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div id="further-information" class="section level2">
<h2 class="hasAnchor">
<a href="#further-information" class="anchor"></a>Further information</h2>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="contrasts---hypothesis-to-be-tested" class="section level1">
<h1 class="hasAnchor">
<a href="#contrasts---hypothesis-to-be-tested" class="anchor"></a>Contrasts - Hypothesis to be tested</h1>
<p>If there are more than two groups, specify the hypothesis to be tested (comparison) using the factor and factor level names.</p>
<p><strong><em>Contrasts to be computed:</em></strong></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p>Examples for parallel Group design (only one factor):</p>
<ul>
<li>Diet - Control</li>
<li>Creme - Control</li>
</ul>
<p>Example for factorial designs:</p>
<ul>
<li>Treatment_Control - Treatment_Creme</li>
<li>AgeGroup_young - AgeGroup_old</li>
<li>Treatment_Control:AgeGroup_young - Treatment_Control:AgeGroup_old</li>
</ul>
<p>Note:</p>
<ul>
<li><p>If one group is used in more comparisons than all other groups, it will benefit the power of the test <a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a> to allocate more samples to this group, since increasing the number of degrees of freedom in this group will benefit many comparisons.</p></li>
<li><p><strong>Limit the number of hypotheses you test</strong>. Multiplicity - each comparison is a test of hypothesis. Therefore, it is required to perform a p-Value adjustment. For instance, using the Bonferroni correction, the obtained p-Values are multiplied by the number of the hypothesis tested. Given two comparisons, a p-Value of 0.03 after Bonferroni adjustment is <span class="math inline">\(0.06\)</span>, i.e., not significant given a test size of <span class="math inline">\(0.05\)</span>. Therefore, do not test all possible hypothesis! For instance, if including the contrast <em>Diet - Creme</em>, it is required to multiply the p-Values with the factor <span class="math inline">\(3\)</span>. i.e., a p-Value of <span class="math inline">\(0.2\)</span> after Bonferroni adjustment is <span class="math inline">\(0.6\)</span> - i.e., not significant.</p></li>
</ul>
<ul>
<li><p>Factorial designs generate many interactions (e.g., Treatment_Control:AgeGroup_young). A <span class="math inline">\(3\times 2\)</span> factorial design generates <span class="math inline">\(6\)</span> groups (unique combinations of factors) which makes it possible to specify many contrasts. Limit the number of hypotheses ideally to few (one or two) per experiment. It is required to adjust the p-Values based on the number of hypotheses examined.</p></li>
<li><p>Typically in factorial designs, tests for differences among main effects (differences between levels within a single factor) and second-order interactions of the factors.</p></li>
<li><p><em>main effects</em> : Treatment_Control - Treatment_Diet</p></li>
<li><p><em>Test for interaction</em> : (Treatment_Control:AgeGroup_young - Treatment_Control:AgeGroup_old) - (Treatment_Diet:AgeGroup_young - Treatment_Diet:AgeGroup_old)</p></li>
</ul>
<div id="more-designs" class="section level2">
<h2 class="hasAnchor">
<a href="#more-designs" class="anchor"></a>More designs</h2>
<ul>
<li>Time Series, e.g., growth curves - require at least 4-5 different well-designed time-points. When less than 4-time points are measured than the time points are typically treated as factors and can be analyzed using our standard pipeline for factorial designs.</li>
</ul>
<p>We might be able to support you with some other hypothesis tests, depending on our availability.</p>
<ul>
<li>Dose-response curves</li>
<li>Survival analysis</li>
</ul>
</div>
</div>
<div id="other-topics" class="section level1">
<h1 class="hasAnchor">
<a href="#other-topics" class="anchor"></a>Other topics</h1>
<p>Example: Normalization is applied to remove systematic differences in protein abundance due to different sample concentrations, or different amount of sample loaded on column. Normalization is important, so that true differentially expressed proteins can be detected.</p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p><span class="math inline">\(~\)</span></p>
<p>Date:</p>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>In statistics, the size of a test is the probability of falsely rejecting the null hypothesis.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>A parallel design, also called a parallel group study, compares two or more treatments. Participants are randomly assigned to either group, treatments are administered, and then the results are compared (statistics how to).<a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p><em>camel case</em> - a typographical convention in which an initial capital is used for the first letter of a word-forming the second element of a closed compound, e.g., PayPal, iPhone, MasterCard.<a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>In statistics, a full factorial experiment is an experiment whose design consists of two or more factors, each with discrete possible values or “levels”, and whose experimental units take on all possible combinations of these levels across all such factors. Such an experiment allows the investigator to study the effect of each factor on the response variable, as well as the effects of interactions between factors on the response variable.<a href="#fnref4" class="footnote-back">↩</a></p></li>
<li id="fn5"><p>Repeated measures design is a research design that involves multiple measures of the same variable taken on the same or matched subjects either under different conditions or over two or more time periods. For instance, repeated measurements are collected in a longitudinal study in which change over time is assessed (Wikipedia).<a href="#fnref5" class="footnote-back">↩</a></p></li>
<li id="fn6"><p>A blocking factor is a factor used to create blocks. It is some variable that has an effect on an experimental outcome, but is itself of no interest. Blocking factors vary wildly depending on the experiment. For example: in human studies age or gender are often used as blocking factors. Blocking factors can be included into the model to describe the variance of the data <a href="https://www.statisticshowto.datasciencecentral.com/blocking-factor/">data science central</a>.<a href="#fnref6" class="footnote-back">↩</a></p></li>
<li id="fn7"><p>The power of a binary hypothesis test is the probability that the test rejects the null hypothesis (H0) when a specific alternative hypothesis (H1) is true. The statistical power ranges from 0 to 1, and as statistical power increases, the probability of making a type II error (wrongly failing to reject the null hypothesis) decreases. <a href="https://en.wikipedia.org/wiki/Power_(statistics)">Wikipedia</a><a href="#fnref7" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Witold Wolski.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
